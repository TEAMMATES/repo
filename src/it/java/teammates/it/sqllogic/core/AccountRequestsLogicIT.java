package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.AccountRequestStatus;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountRequestsLogic;
import teammates.storage.sqlapi.AccountRequestsDb;
import teammates.storage.sqlentity.AccountRequest;

/**
 * SUT: {@link AccountRequestsLogic}.
 */
public class AccountRequestsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountRequestsLogic accountRequestsLogic = AccountRequestsLogic.inst();

    @Test
    public void testGetAccountRequest_nonExistentAccountRequest_returnsNull() {
        UUID id = UUID.randomUUID();
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertNull(actualAccountRequest);
    }

    @Test
    public void testGetAccountRequest_existingAccountRequest_getsSuccessfully() throws InvalidParametersException {
        AccountRequest expectedAccountRequest =
                new AccountRequest("test@gmail.com", "name", "institute", AccountRequestStatus.PENDING, "comments");
        UUID id = expectedAccountRequest.getId();
        accountRequestsLogic.createAccountRequest(expectedAccountRequest);
        AccountRequest actualAccountRequest = accountRequestsLogic.getAccountRequest(id);
        assertEquals(expectedAccountRequest, actualAccountRequest);
    }

    @Test
    public void testResetAccountRequest()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {

        ______TS("success: create account request and update registeredAt field");

        String name = "name lee";
        String email = "email@gmail.com";
        String institute = "institute";
        AccountRequestStatus status = AccountRequestStatus.PENDING;
        String comments = "comments";

        AccountRequest toReset = accountRequestsLogic.createAccountRequest(name, email, institute, status, comments);
        AccountRequestsDb accountRequestsDb = AccountRequestsDb.inst();

        toReset.setRegisteredAt(Instant.now());
        toReset = accountRequestsDb.getAccountRequest(email, institute);

        assertNotNull(toReset);
        assertNotNull(toReset.getRegisteredAt());

        ______TS("success: reset account request that already exists");

        AccountRequest resetted = accountRequestsLogic.resetAccountRequest(email, institute);

        assertNull(resetted.getRegisteredAt());

        ______TS("success: test delete account request");

        accountRequestsLogic.deleteAccountRequest(email, institute);

        assertNull(accountRequestsLogic.getAccountRequest(email, institute));

        ______TS("failure: reset account request that does not exist");

        assertThrows(EntityDoesNotExistException.class,
                () -> accountRequestsLogic.resetAccountRequest(name, institute));
    }
}
