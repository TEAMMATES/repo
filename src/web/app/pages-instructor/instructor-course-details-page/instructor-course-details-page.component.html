<div *ngIf="courseDetails">
  <h1>Course Details</h1>
  <div class="card">
    <div class="card-body fill-plain">
      <div class="form">
        <div class="form-group row">
          <label class="col-sm-3 text-right">Course ID:</label>
          <div class="col-sm-6">
            <p class="form-control-static">{{courseDetails.course.id}}</p>
          </div>
        </div>
        <div class="form-group row">
          <label class="col-sm-3 text-right">Course name:</label>
          <div class="col-sm-6" id="coursename">
            <p class="form-control-static">{{courseDetails.course.name}}</p>
          </div>
        </div>
        <div class="form-group row">
          <label class="col-sm-3 text-right">Sections:</label>
          <div class="col-sm-6">
            <p class="form-control-static">{{courseDetails.stats.sectionsTotal}}</p>
          </div>
        </div>
        <div class="form-group row">
          <label class="col-sm-3 text-right">Teams:</label>
          <div class="col-sm-6">
            <p class="form-control-static">{{courseDetails.stats.teamsTotal}}</p>
          </div>
        </div>
        <div class="form-group row">
          <label class="col-sm-3 text-right">Total students:</label>
          <div class="col-sm-6">
            <p class="form-control-static">{{courseDetails.stats.studentsTotal}}</p>
          </div>
        </div>
      </div>
      <div class="form-group row">
        <label class="col-sm-3 control-label text-right">Instructors:</label>
        <div class="col-sm-6">
          <div *ngIf="instructors">
            <div *ngFor="let instructor of instructors">
              <ng-container *ngIf="instructor.role; else showRole">
                Co-owner: {{instructor.name}} ({{instructor.email}})
              </ng-container>
              <ng-template #showRole >
                <p>
                  {{instructor.role}} {{instructor.name}} ({{instructor.email}})
                </p>
              </ng-template>
            </div>
          </div>
        </div>
      </div>
      <div *ngIf="courseDetails.stats.studentsTotal;">
        <div class="form-group">
          <div class="text-center">
            <button type="button" class="btn btn-primary" (click)="openModal(remindStudentsModal)" value="Remind Students to Join">
              <i class="fa fa-envelope"></i>
              Remind Students to Join
            </button>
            <button type="submit" class="btn btn-primary" name="fruploaddownloadbtn">
              <i class="fa fa-download"></i>
              Download Student List
            </button>
            <button type="button" class="btn btn-danger" (click)="openModal(deleteAllStudentsModal)" value="Delete All Students">
              <i class="fa fa-trash"></i>
              Delete All Students
            </button>
            <div>
              <span>Non-English characters not displayed properly in the downloaded file?
                <span class="btn-link" (click)="openModal(studentTableModal)">
                  Click here.
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>