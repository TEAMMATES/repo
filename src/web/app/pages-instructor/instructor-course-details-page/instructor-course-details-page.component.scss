.card {
  margin-bottom: 20px;
}

.form-group {
  margin-bottom: 0.5rem;
}

.fill-plain {
  background-color: #EAEFF5;
}

button {
  margin-right: 5px;
}

label {
  font-weight: bold;
}

.btn-link:hover {
  cursor: pointer;
}

.student-table-margin {
  margin-top: 50px;
}
