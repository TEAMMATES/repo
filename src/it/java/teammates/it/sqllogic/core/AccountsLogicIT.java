package teammates.it.sqllogic.core;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

import org.testng.annotations.Test;

import teammates.common.datatransfer.NotificationStyle;
import teammates.common.datatransfer.NotificationTargetUser;
import teammates.common.exception.EntityAlreadyExistsException;
import teammates.common.exception.EntityDoesNotExistException;
import teammates.common.exception.InvalidParametersException;
import teammates.it.test.BaseTestCaseWithSqlDatabaseAccess;
import teammates.sqllogic.core.AccountsLogic;
import teammates.sqllogic.core.NotificationsLogic;
import teammates.storage.sqlapi.AccountsDb;
import teammates.storage.sqlentity.Account;
import teammates.storage.sqlentity.Notification;
import teammates.storage.sqlentity.ReadNotification;

/**
 * SUT: {@link AccountsLogic}.
 */
public class AccountsLogicIT extends BaseTestCaseWithSqlDatabaseAccess {

    private AccountsLogic accountsLogic = AccountsLogic.inst();
    private NotificationsLogic notificationsLogic = NotificationsLogic.inst();

    private AccountsDb accountsDb = AccountsDb.inst();

    @Test
    public void testUpdateReadNotifications()
            throws EntityAlreadyExistsException, InvalidParametersException, EntityDoesNotExistException {
        ______TS("success: mark notification as read");
        Account account = new Account("google-id", "name", "email@teammates.com");
        Notification notification = new Notification(Instant.parse("2011-01-01T00:00:00Z"),
                Instant.parse("2099-01-01T00:00:00Z"), NotificationStyle.DANGER, NotificationTargetUser.GENERAL,
                "A deprecation note", "<p>Deprecation happens in three minutes</p>");
        accountsDb.createAccount(account);
        notificationsLogic.createNotification(notification);

        String googleId = account.getGoogleId();
        UUID notificationId = notification.getId();
        accountsLogic.updateReadNotifications(googleId, notificationId, notification.getEndTime());

        Account actualAccount = accountsDb.getAccountByGoogleId(googleId);
        List<ReadNotification> accountReadNotifications = actualAccount.getReadNotifications();
        assertEquals(1, accountReadNotifications.size());
        assertSame(actualAccount, accountReadNotifications.get(0).getAccount());
        assertSame(notification, accountReadNotifications.get(0).getNotification());
    }
}
