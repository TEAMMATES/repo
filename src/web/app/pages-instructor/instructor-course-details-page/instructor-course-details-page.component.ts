import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';
import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
import { saveAs } from 'file-saver';
import { ClipboardService } from 'ngx-clipboard';
import { HttpRequestService } from '../../../services/http-request.service';
import { NavigationService } from '../../../services/navigation.service';
import { StatusMessageService } from '../../../services/status-message.service';
import { ErrorMessageOutput, MessageOutput } from '../../message-output';
import { StudentListSectionData } from '../student-list/student-list-section-data';

interface CourseAttributes {
  id: string;
  name: string;
}

interface CourseStats {
  sectionsTotal: string;
  teamsTotal: string;
  studentsTotal: string;
}

interface CourseDetailsBundle {
  course: CourseAttributes;
  stats: CourseStats;
}

interface InstructorAttributes {
  googleId: string;
  name: string;
  email: string;
  key: string;
  role: string;
  displayedName: string;
  isArchived: boolean;
  isDisplayedToStudents: boolean;
}

interface CourseInfo {
  courseDetails: CourseDetailsBundle;
  currentInstructor: InstructorAttributes;
  instructors: InstructorAttributes[];
  sections: StudentListSectionData[];
  hasSection: boolean;
  studentListHtmlTableAsString: string;
}

interface RedirectInfo {
  redirectUrl: string;
  statusMessage: string;
}

/**
 * Instructor course details page.
 */
@Component({
  selector: 'tm-instructor-course-details-page',
  templateUrl: './instructor-course-details-page.component.html',
  styleUrls: ['./instructor-course-details-page.component.scss'],
})
export class InstructorCourseDetailsPageComponent implements OnInit {

  user: string = '';
  courseDetails?: CourseDetailsBundle;
  currentInstructor?: InstructorAttributes;
  instructors?: InstructorAttributes[] = [];
  sections?: StudentListSectionData[] = [];
  studentListHtmlTableAsString?: String = '';

  constructor(private route: ActivatedRoute, private router: Router,
              private clipboardService: ClipboardService,
              private httpRequestService: HttpRequestService,
              private statusMessageService: StatusMessageService,
              private ngbModal: NgbModal, private navigationService: NavigationService) { }

  ngOnInit(): void {
    this.route.queryParams.subscribe((queryParams: any) => {
      this.user = queryParams.user;
      this.loadCourseDetails(queryParams.courseid);
    });
  }

  /**
   * Loads the course's details based on the given course ID and email.
   */
  loadCourseDetails(courseid: string): void {
    const paramMap: { [key: string]: string } = { courseid };
    this.httpRequestService.get('/courses/details', paramMap).subscribe((resp: CourseInfo) => {
      this.courseDetails = resp.courseDetails;
      this.currentInstructor = resp.currentInstructor;
      this.instructors = resp.instructors;
      this.sections = resp.sections;
      this.studentListHtmlTableAsString = resp.studentListHtmlTableAsString;

      if (!this.courseDetails) {
        this.statusMessageService.showErrorMessage('Error retrieving course details');
      }
    }, (resp: ErrorMessageOutput) => {
      this.statusMessageService.showErrorMessage(resp.error.message);
    });
  }

  /**
   * Automatically copy the text content provided.
   */
  copyContent(text: string): void {
    this.clipboardService.copyFromContent(text);
  }

  /**
   * Open the modal for different buttons and link.
   */
  openModal(content: any): void {
    this.ngbModal.open(content);
  }

  /**
   * Delete all the students in a course.
   */
  deleteAllStudentsFromCourse(courseId: string): void {
    const paramsMap: { [key: string]: string } = {
      user: this.user,
      courseid: courseId,
    };

    this.httpRequestService.delete('/courses/details/deleteAllStudents', paramsMap)
      .subscribe((resp: MessageOutput) => {
        this.navigationService.navigateWithSuccessMessage(this.router,
            `/web/instructor/courses/details?courseid=${courseId}`,
            resp.message);
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorMessage(resp.error.message);
      });
  }

  /**
   * Download all the students from a course.
   */
  downloadAllStudentsFromCourse(courseId: string): void {
    const paramsMap: { [key: string]: string } = { courseid: courseId };
    this.httpRequestService.get('/courses/details/downloadAllStudents', paramsMap, 'text')
      .subscribe((resp: string) => {
        const filename: string = `${courseId.concat('_studentList')}.csv`;
        const blob: any = new Blob([resp], { type: 'text/csv' });
        saveAs(blob, filename);
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorMessage(resp.error.message);
      });
  }

  /**
   * Remind all unjoined students in a course.
   */
  remindAllStudentsFromCourse(courseId: string): void {
    const paramsMap: { [key: string]: string } = {
      user: this.user,
      courseid: courseId,
    };

    this.httpRequestService.post('/courses/details/remindAllStudents', paramsMap)
      .subscribe((resp: RedirectInfo) => {
        this.navigationService.navigateWithSuccessMessage(this.router, resp.redirectUrl, resp.statusMessage);
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorMessage(resp.error.message);
      });
  }
}
