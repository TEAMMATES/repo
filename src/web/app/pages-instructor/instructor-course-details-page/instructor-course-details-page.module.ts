import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';
import { AjaxLoadingModule } from '../../components/ajax-loading/ajax-loading.module';
import { AjaxPreloadModule } from '../../components/ajax-preload/ajax-preload.module';
import { StudentListModule } from '../../components/student-list/student-list.module';
import { TeammatesCommonModule } from '../../components/teammates-common/teammates-common.module';
import { InstructorCourseDetailsPageComponent } from './instructor-course-details-page.component';

const routes: Routes = [
  {
    path: '',
    component: InstructorCourseDetailsPageComponent,
  },
];

/**
 * Module for instructor course details page.
 */
@NgModule({
  declarations: [
    InstructorCourseDetailsPageComponent,
  ],
  exports: [
    InstructorCourseDetailsPageComponent,
  ],
  imports: [
    CommonModule,
    TeammatesCommonModule,
    RouterModule.forChild(routes),
    StudentListModule,
    AjaxLoadingModule,
    AjaxPreloadModule,
  ],
})
export class InstructorCourseDetailsPageModule { }
