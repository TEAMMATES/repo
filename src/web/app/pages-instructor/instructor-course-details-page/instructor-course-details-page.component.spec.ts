import { HttpClientTestingModule } from '@angular/common/http/testing';
import { Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { InstructorCourseDetailsPageComponent } from './instructor-course-details-page.component';

@Component({ selector: 'tm-student-list', template: '' })
class StudentListStubComponent {
  @Input() courseId: string = '';
  @Input() useGrayHeading: boolean = true;
  @Input() sections: Object[] = [];
  @Input() fromCourseDetailsPage: boolean = true;
}

describe('InstructorCourseDetailsPageComponent', () => {
  let component: InstructorCourseDetailsPageComponent;
  let fixture: ComponentFixture<InstructorCourseDetailsPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        InstructorCourseDetailsPageComponent,
        StudentListStubComponent,
      ],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorCourseDetailsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
