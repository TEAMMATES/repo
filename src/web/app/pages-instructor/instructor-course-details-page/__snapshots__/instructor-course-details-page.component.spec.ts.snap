// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorCourseDetailsPageComponent should snap with a course with one co-owner and no students, and populated course student list 1`] = `
<tm-instructor-course-details-page
  clipboardService={[Function ClipboardService]}
  courseDetails={[Function Object]}
  courseService={[Function CourseService]}
  courseStudentListAsCsv={[Function String]}
  instructorService={[Function InstructorService]}
  instructors={[Function Array]}
  isAjaxSuccess={[Function Boolean]}
  loading={[Function Boolean]}
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  students={[Function Array]}
>
  <div
    class="text-center"
  >
    <h1>
      Course Details
    </h1>
    <div
      class="card"
    >
      <div
        class="card-body fill-plain"
      >
        <div
          class="form"
        >
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Course ID:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                CS101
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Course name:
            </label>
            <div
              class="col-sm-6 text-sm-left"
              id="coursename"
            >
              <p
                class="form-control-static"
              >
                Introduction to CS
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Sections:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Teams:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Total students:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
        </div>
        <div
          class="form-group row"
        >
          <label
            class="col-sm-3 text-sm-right control-label"
          >
            Instructors:
          </label>
          <div
            class="col-sm-6 text-sm-left"
          >
            
            <div>
               Co-owner: Hodor (hodor@gmail.com) 
            </div>
          </div>
        </div>
        
      </div>
    </div>
    
    <div
      class="alert alert-warning"
    >
      <i
        class="fa fa-exclamation-circle text-muted"
      />
       There are no students in this course. Click 
      <a
        href="/web/instructor/courses/enroll?courseid=CS101"
        ng-reflect-query-params="[object Object]"
        ng-reflect-router-link="/web/instructor/courses/enroll"
        routerlink="/web/instructor/courses/enroll"
      >
        here
      </a>
       to enroll students. 
    </div>
    
    
    
    
  </div>
</tm-instructor-course-details-page>
`;

exports[`InstructorCourseDetailsPageComponent should snap with a course with one co-owner and one student, and ajax failure 1`] = `
<tm-instructor-course-details-page
  clipboardService={[Function ClipboardService]}
  courseDetails={[Function Object]}
  courseService={[Function CourseService]}
  courseStudentListAsCsv=""
  instructorService={[Function InstructorService]}
  instructors={[Function Array]}
  isAjaxSuccess="false"
  loading="false"
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  students={[Function Array]}
>
  <div
    class="text-center"
  >
    <h1>
      Course Details
    </h1>
    <div
      class="card"
    >
      <div
        class="card-body fill-plain"
      >
        <div
          class="form"
        >
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Course ID:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                CS101
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Course name:
            </label>
            <div
              class="col-sm-6 text-sm-left"
              id="coursename"
            >
              <p
                class="form-control-static"
              >
                Introduction to CS
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Sections:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Teams:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Total students:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
        </div>
        <div
          class="form-group row"
        >
          <label
            class="col-sm-3 text-sm-right control-label"
          >
            Instructors:
          </label>
          <div
            class="col-sm-6 text-sm-left"
          >
            
            <div>
               Co-owner: Bran (bran@gmail.com) 
            </div>
          </div>
        </div>
        
        <div>
          <div
            class="form-group"
          >
            <div
              class="text-center"
            >
              <button
                class="btn btn-primary mb-2"
                type="button"
                value="Remind Students to Join"
              >
                <i
                  class="fa fa-envelope"
                />
                 Remind Students to Join 
              </button>
              <button
                class="btn btn-primary mb-2"
                type="button"
              >
                <i
                  class="fa fa-download"
                />
                 Download Student List 
              </button>
              <button
                class="btn btn-danger mb-2"
                type="button"
                value="Delete All Students"
              >
                <i
                  class="fa fa-trash"
                />
                 Delete All Students 
              </button>
              <div>
                <span>
                  Non-English characters not displayed properly in the downloaded file? 
                  <span
                    class="btn-link"
                  >
                     Click here. 
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    
    <div
      class="card-body p-0"
    >
      <div
        class="card-header text-dark-blue bg-light-blue font-weight-bold"
      >
        [CS101]
      </div>
      <div
        class="card-body p-0"
      >
        <tm-student-list
          ng-reflect-course-id="CS101"
          ng-reflect-enable-remind-button="true"
          ng-reflect-students="[object Object]"
          ng-reflect-use-gray-heading="true"
        >
          <div
            class="table-responsive"
          >
            <table
              class="table table-bordered table-striped m-0"
            >
              <thead
                class="thead-gray"
                ng-reflect-ng-class="[object Object]"
              >
                <tr>
                  <th>
                    Photo
                  </th>
                  
                  <th
                    class="sortable-header"
                  >
                     Section 
                    
                    <span>
                      <i
                        class="fas fa-sort"
                      />
                    </span>
                    
                    
                  </th>
                  <th
                    class="sortable-header"
                  >
                     Team 
                    
                    <span>
                      <i
                        class="fas fa-sort"
                      />
                    </span>
                    
                    
                  </th>
                  <th
                    class="sortable-header"
                  >
                     Student Name 
                    
                    <span>
                      <i
                        class="fas fa-sort"
                      />
                    </span>
                    
                    
                  </th>
                  <th
                    class="sortable-header"
                  >
                     Status 
                    
                    <span>
                      <i
                        class="fas fa-sort"
                      />
                    </span>
                    
                    
                  </th>
                  <th
                    class="sortable-header"
                  >
                     Email 
                    
                    <span>
                      <i
                        class="fas fa-sort"
                      />
                    </span>
                    
                    
                  </th>
                  <th>
                    Action(s)
                  </th>
                </tr>
              </thead>
              <tbody>
                
                <tr
                  s=""
                >
                  <td
                    class="align-center"
                  >
                    <tm-view-photo-popover
                      ng-reflect-is-view-photo-link-in-popover="false"
                      ng-reflect-photo-url="http://localhost:8080/webapi/s"
                    >
                      
                      
                      
                      <a
                        href="/"
                        ng-reflect-query-params-handling="preserve"
                        queryparamshandling="preserve"
                        role="button"
                      >
                         View Photo 
                      </a>
                      
                      
                      
                    </tm-view-photo-popover>
                  </td>
                  
                  <td>
                    Tutorial Group 1
                  </td>
                  <td>
                    Team 1
                  </td>
                  <td>
                    Jamie
                  </td>
                  <td
                    class="align-center"
                  >
                    Yet to Join
                  </td>
                  <td>
                    jamie@gmail.com
                  </td>
                  <td
                    class="no-print align-center"
                  >
                    
                    
                    <a
                      class="btn btn-light btn-sm mr-2"
                      href="/web/instructor/courses/student/details?courseid=CS101&studentemail=jamie@gmail.com"
                      ng-reflect-klass="btn btn-light btn-sm mr-2"
                      ng-reflect-ng-class="[object Object]"
                      ng-reflect-ngb-tooltip="View the details of the studen"
                      ng-reflect-query-params="[object Object]"
                      ng-reflect-router-link="/web/instructor/courses/studen"
                      ng-reflect-target="_blank"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                       View
                    </a>
                    
                    
                    <a
                      class="btn btn-light btn-sm mr-2"
                      href="/web/instructor/courses/student/edit?courseid=CS101&studentemail=jamie@gmail.com"
                      ng-reflect-klass="btn btn-light btn-sm mr-2"
                      ng-reflect-ng-class="[object Object]"
                      ng-reflect-ngb-tooltip="Use this to edit the details o"
                      ng-reflect-query-params="[object Object]"
                      ng-reflect-router-link="/web/instructor/courses/studen"
                      ng-reflect-target="_blank"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                       Edit
                    </a>
                    
                    
                    
                    
                    
                    <button
                      class="btn btn-light btn-sm mr-2"
                      ng-reflect-klass="btn btn-light btn-sm mr-2"
                      ng-reflect-ng-class="[object Object]"
                      ng-reflect-ngb-tooltip="Email an invitation to the stu"
                    >
                      Send Invite
                    </button>
                    
                    
                    
                    <button
                      class="btn btn-light btn-sm mr-2"
                      ng-reflect-klass="btn btn-light btn-sm mr-2"
                      ng-reflect-ng-class="[object Object]"
                      ng-reflect-ngb-tooltip="Delete the student and the cor"
                    >
                      Delete
                    </button>
                    
                    
                    <a
                      class="btn btn-light btn-sm mr-2"
                      href="/web/instructor/students/records?courseid=CS101&studentemail=jamie@gmail.com"
                      ng-reflect-klass="btn btn-light btn-sm mr-2"
                      ng-reflect-ng-class="[object Object]"
                      ng-reflect-ngb-tooltip="View all data about this stude"
                      ng-reflect-query-params="[object Object]"
                      ng-reflect-router-link="/web/instructor/students/recor"
                      ng-reflect-target="_blank"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                       All Records
                    </a>
                    
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </tm-student-list>
      </div>
    </div>
    
    
    
    
  </div>
</tm-instructor-course-details-page>
`;

exports[`InstructorCourseDetailsPageComponent should snap with default fields 1`] = `
<tm-instructor-course-details-page
  clipboardService={[Function ClipboardService]}
  courseDetails={[Function Object]}
  courseService={[Function CourseService]}
  courseStudentListAsCsv=""
  instructorService={[Function InstructorService]}
  instructors={[Function Array]}
  isAjaxSuccess={[Function Boolean]}
  loading="false"
  navigationService={[Function NavigationService]}
  ngbModal={[Function NgbModal]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  statusMessageService={[Function StatusMessageService]}
  studentService={[Function StudentService]}
  students={[Function Array]}
>
  <div
    class="text-center"
  >
    <h1>
      Course Details
    </h1>
    <div
      class="card"
    >
      <div
        class="card-body fill-plain"
      >
        <div
          class="form"
        >
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Course ID:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Course name:
            </label>
            <div
              class="col-sm-6 text-sm-left"
              id="coursename"
            >
              <p
                class="form-control-static"
              >
                
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Sections:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                0
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Teams:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                0
              </p>
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-3 text-sm-right"
            >
              Total students:
            </label>
            <div
              class="col-sm-6 text-sm-left"
            >
              <p
                class="form-control-static"
              >
                0
              </p>
            </div>
          </div>
        </div>
        <div
          class="form-group row"
        >
          <label
            class="col-sm-3 text-sm-right control-label"
          >
            Instructors:
          </label>
          <div
            class="col-sm-6 text-sm-left"
          >
            
          </div>
        </div>
        
      </div>
    </div>
    
    <div
      class="alert alert-warning"
    >
      <i
        class="fa fa-exclamation-circle text-muted"
      />
       There are no students in this course. Click 
      <a
        href="/web/instructor/courses/enroll?courseid="
        ng-reflect-query-params="[object Object]"
        ng-reflect-router-link="/web/instructor/courses/enroll"
        routerlink="/web/instructor/courses/enroll"
      >
        here
      </a>
       to enroll students. 
    </div>
    
    
    
    
  </div>
</tm-instructor-course-details-page>
`;
